controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -171,9 +171,10 @@ public OperationContext createContext(String scope, String name) {
 
     @Override
     public CompletableFuture<State> getState(final String scope, final String name,
+                                             final boolean ignoreCached,
```

Comment:
Could we add some comments her to explain when we are using the cached value and when not?

Reply:
Done.
we use ignoreCached whenever we intend to update the state. For this we get the latest value from the store and then use it for updates if transitions are allowed.